css/selectors/state.json

```
@@ -0,0 +1,49 @@
+{
+  "css": {
+    "selectors": {
+      "state": {
+        "__compat": {
+          "description": "Custom state pseudo-class selector (<code>:state()</code>)",
+          "mdn_url": "https://developer.mozilla.org/docs/Web/CSS/:state",
+          "spec_url": "https://html.spec.whatwg.org/multipage/semantics-other.html#selector-custom",
+          "support": {
+            "chrome": {
+              "version_added": false,
+              "notes": "From version 90 Chromium supports custom state selection using a dashed-ident (such as <code>:--foo</code>) instead of <code>:state()</code>."
+            },
+            "chrome_android": "mirror",
+            "edge": "mirror",
+            "firefox": {
+              "version_added": "121",
+              "flags": [
+                {
+                  "type": "preference",
+                  "name": "dom.element.customstateset.enabled",
+                  "value_to_set": "true"
+                }
+              ]
+            },
+            "firefox_android": "mirror",
+            "ie": {
+              "version_added": false
+            },
+            "oculus": "mirror",
+            "opera": "mirror",
+            "opera_android": "mirror",
+            "safari": {
+              "version_added": false
+            },
```

Mirror CustomStateSet - this was added in https://developer.apple.com/documentation/safari-technology-preview-release-notes/stp-release-187#New-Features

```suggestion
            "safari": {
              "version_added": "preview",
              "flags": [
                {
                  "name": "CustomStateSet",
                  "value_to_set": "true",
                  "type": "preference"
                }
              ]
            },
```